### PR TITLE
feat(data-transfer): data import supports file directory

### PR DESCRIPTION
**Pull request by persistafei**
What type of PR is this?
type-feature

What this PR does / why we need it:
data import now supports choosing file directory

Which issue(s) this PR fixes:
Fixes #

Special notes for your reviewer:
Additional documentation e.g., usage docs, etc.:

### CODE REVIEW

File: server/plugins/task-plugin-api/src/main/java/com/oceanbase/odc/plugin/task/api/datatransfer/dumper/ExportOutput.java [hunk 1]
```diff
@@ -102,6 +107,17 @@ public void toFolder(@NonNull File folder) throws IOException {
         if (!folder.exists()) {
             FileUtils.forceMkdir(folder);
         }
+
+        if (inputDir != null) {
+            // if input file is a dir, try to use symbolic link, avoid copy files if possible
+            try {
+                Files.createSymbolicLink(this.inputDir, folder.toPath());
```
Comment (yizhouxw) [inline server/plugins/task-plugin-api/src/main/java/com/oceanbase/odc/plugin/task/api/datatransfer/dumper/ExportOutput.java:114]: when the symbolic link will be unlinked ?

is it possibile to read from from origin directory?

File: server/plugins/task-plugin-api/src/main/java/com/oceanbase/odc/plugin/task/api/datatransfer/dumper/ExportOutput.java [hunk 2]
```diff
@@ -66,12 +66,16 @@ public class ExportOutput {
     private final BinaryFile<Manifest> manifest;
     private final BinaryFile<List<?>> checkpoints;
     private final List<DumpDBObject> dumpDbObjects;
+    private boolean isZip;
```
Comment (yizhouxw) [inline server/plugins/task-plugin-api/src/main/java/com/oceanbase/odc/plugin/task/api/datatransfer/dumper/ExportOutput.java:69]: no `is` prefix for boolean field, use `zip` straightly

File: server/odc-service/src/main/java/com/oceanbase/odc/service/datatransfer/loader/PlSqlFormat.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import java.util.HashSet;
+import java.util.Set;
+
+import com.oceanbase.odc.common.util.StringUtils;
+
+/**
+ * @author youshu
+ * @date 2024/8/27
+ */
+public class PlSqlFormat {
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/datatransfer/loader/PlSqlFormat.java:27]: PLSqlDeveloperExportFormat ?

1. it is the file format used by PL/SQL Developer tool exported, not PL/SQL languange identified.
2. Sql should be SQL better indeed, however too much Sql used in odc already..

File: server/plugins/task-plugin-api/src/main/java/com/oceanbase/odc/plugin/task/api/datatransfer/dumper/ExportOutput.java (hunk 1 above)
Comment (persistafei) [reply server/plugins/task-plugin-api/src/main/java/com/oceanbase/odc/plugin/task/api/datatransfer/dumper/ExportOutput.java:114]: after data transfer task finished, either success or failure, the /datatransfer/${seq}  temp directory will be cleaned, and files will be unlinked.

if read from origin directory, log files and some hidden files will be written to that place, which might cause some trouble for user.

File: server/plugins/task-plugin-api/src/main/java/com/oceanbase/odc/plugin/task/api/datatransfer/dumper/ExportOutput.java (hunk 2 above)
Comment (persistafei) [reply server/plugins/task-plugin-api/src/main/java/com/oceanbase/odc/plugin/task/api/datatransfer/dumper/ExportOutput.java:69]: OK

File: server/odc-service/src/main/java/com/oceanbase/odc/service/datatransfer/loader/PlSqlFormat.java (hunk 3 above)
Comment (persistafei) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/datatransfer/loader/PlSqlFormat.java:27]: OK

**Review by yizhouxw — DISMISSED** [review]
lgtm

**Review by yizhouxw — APPROVED** [review]

**Pull request closed without merging**